satellite/metainfo/metainfo.go

```diff
@@ -506,3 +509,34 @@ func CreatePath(ctx context.Context, projectID uuid.UUID, segmentIndex int64, bu
 	}
 	return storj.JoinPaths(entries...), nil
 }
+
+// checks if bucket has any pointers(entries)
+func (endpoint *Endpoint) checkBucketPointers(ctx context.Context, req *pb.SegmentInfoRequest) (*pb.SegmentInfoResponse, error) {
+	// not_found error indicates connectory id set but no entry exists
+	resp, err := endpoint.SegmentInfo(ctx, req)
```

Comment:
It doesn't check bucket entries but if segment exists. also don't reuse external endpoint methods, use logic from services.